easybuild/easyconfigs/s/scikit-learn/scikit-learn-0.20.3-intel-2019a-Python-3.7.2.eb

```diff
@@ -0,0 +1,38 @@
+easyblock = 'PythonPackage'
+
+name = 'scikit-learn'
+version = '0.20.3'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://scikit-learn.org/stable/index.html'
+description = """Scikit-learn integrates machine learning algorithms in the tightly-knit scientific Python world,
+building upon numpy, scipy, and matplotlib. As a machine-learning module,
+it provides versatile tools for data mining and analysis in any field of science and engineering.
+It strives to be simple and efficient, accessible to everybody, and reusable in various contexts."""
+
+toolchain = {'name': 'intel', 'version': '2019a'}
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+checksums = ['c503802a81de18b8b4d40d069f5e363795ee44b1605f38bc104160ca3bfe2c41']
+
+dependencies = [
+    ('Python', '3.7.2'),
+    ('SciPy-bundle', '2019.03'),  # required for numpy/scipy
+    ('matplotlib', '3.0.3', versionsuffix),
```

Comment:
Do you need to depend on matplotlib? I see no need. It's perfectly useable without, and, without, we can use multi_deps. pip doens't consider matplotlib a dependency.

From examples, it only looks like you can plot the results from scikit-learn with matplotlib. But, you can say that about everything.

Reply:
Well, we can definitely consider dropping `matplotlib` as a dep.

But it won't do us much good in the long run anyway w.r.t. `multi_deps`, since the next `scikit-learn` won't support Python 2.x anyway?

Reply:
I'm all for phasing out old crusty python2, but no need to do so unnecessarily?

Either way, we were working on some other packages which depend on scikit-learn, and it wouldn't really need to pull in the entire graphics stack as a dependency unnecessarily.

If you want to keep it python3 only, I wouldn't care to much, the stuff that I'm working on is python3 only anyway.

Reply:
Yeah, ok, I see your point.

`matplotlib` removed as dep and switched to using `multi_deps` in 06ff48c15